loader/loader.go

```diff
@@ -377,20 +377,44 @@ func (l *Loader) convertMessage(tspec *ast.TypeSpec) (*desc.DescriptorProto, err
 		if mtype, ok := ftype.(*types.Map); ok {
 			ptype = desc.FieldDescriptorProto_TYPE_MESSAGE
 			plabel = desc.FieldDescriptorProto_LABEL_REPEATED
-			tname, msgNestedType = l.convertMap(tspec.Name.Name, field.Names[0].Name, mtype)
+			tname, msgNestedType = l.convertMap(tspec.Name.Name, fieldName, mtype)
 			msg.NestedType = append(msg.NestedType, msgNestedType)
 		} else {
 			ptype, plabel, tname = l.convertType(ftype)
 		}
 		if ptype == 0 {
 			return nil, fmt.Errorf("unsupported field type: %v", ftype)
 		}
+
+		// Check that the struct field has a tag. We currently
+		// require all struct fields to have a tag; this is used
+		// to assign the position number for a field, ie: `pb:"1"`
+		if field.Tag == nil {
+			return nil, fmt.Errorf("missing required tag on %s", fieldName)
+		}
+		str, err := strconv.Unquote(field.Tag.Value)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unquote tag on %s", fieldName)
+		}
+		tag := reflect.StructTag(str)
+		// TODO: record the position numbers used so we can return an
+		// error if position number is used more than once? This would
+		// also allow us to automatically assign fields a position
+		// number if it is missing one.
+		num, err := protoNumber(tag)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert tag to number on %s: %v", fieldName, err)
+		}
+
+		// TODO: We aren't currently setting the JsonName field, not sure what we want
+		// to set them to? Possibly pull out a json tag or something?
 		msg.Field = append(msg.Field, &desc.FieldDescriptorProto{
-			Name:     proto.String(field.Names[0].Name),
-			Number:   protoNumber(field.Tag),
+			Name:     proto.String(fieldName),
+			Number:   num,
 			TypeName: &tname,
 			Type:     &ptype,
 			Label:    &plabel,
+			JsonName: nil,
```

Comment:
remove?

Reply:
Argh, that was in preparation for the json fields, I'll remove.